dist/picturefill.js

```diff
@@ -47,678 +47,705 @@ window.matchMedia || (window.matchMedia = function() {
+		 * this moves srcset's value to memory for later use and removes the attr
+		 */
+		pf.dodgeSrcset = function( img ) {
+			if ( img.srcset ) {
+				img[ pf.ns ].srcset = img.srcset;
+				img.srcset = "";
+				img.setAttribute( "data-pfsrcset", img[ pf.ns ].srcset );
+			}
+		};
 
-			style.zoom = currentZoom;
-		}
-	};
+		// Accept a source or img element and process its srcset and sizes attrs
+		pf.processSourceSet = function( el ) {
+			var srcset = el.getAttribute( "srcset" ),
+				sizes = el.getAttribute( "sizes" ),
+				candidates = [];
 
-	pf.setIntrinsicSize = (function() {
-		var urlCache = {};
-		var setSize = function( picImg, width, res ) {
-			picImg.setAttribute( "width", parseInt(width / res, 10) );
-		};
-		return function( picImg, bestCandidate ) {
-			var img;
-			if ( !picImg[ pf.ns ] ) {
-				return;
-			}
-			if ( picImg[ pf.ns ].dims === undefined ) {
-				picImg[ pf.ns].dims = picImg.getAttribute("width") || picImg.getAttribute("height");
+			// if it's an img element, use the cached srcset property (defined or not)
+			if ( el.nodeName.toUpperCase() === "IMG" && el[ pf.ns ] && el[ pf.ns ].srcset ) {
+				srcset = el[ pf.ns ].srcset;
 			}
-			if ( picImg[ pf.ns].dims ) { return; }
-
-			if ( urlCache[bestCandidate.url] ) {
-				setSize( picImg, urlCache[bestCandidate.url], bestCandidate.resolution );
-			} else {
-				img = doc.createElement( "img" );
-				img.onload = function() {
-					urlCache[bestCandidate.url] = img.width;
-					if ( picImg.src === bestCandidate.url ) {
-						setSize( picImg, urlCache[bestCandidate.url], bestCandidate.resolution );
-					}
-					picImg = null;
-					img.onload = null;
-					img = null;
-				};
-				img.src = bestCandidate.url;
+
+			if ( srcset ) {
+				candidates = pf.getCandidatesFromSourceSet( srcset, sizes );
 			}
+			return candidates;
 		};
-	})();
-
-	pf.applyBestCandidate = function( candidates, picImg ) {
-		var candidate,
-			length,
-			bestCandidate;
 
-		candidates.sort( pf.ascendingSort );
+		// pf.backfaceVisibilityFix = function( picImg ) {
+		// 	// See: https://github.com/scottjehl/picturefill/issues/332
+		// 	var style = picImg.style || {},
+		// 		WebkitBackfaceVisibility = "webkitBackfaceVisibility" in style,
+		// 		currentZoom = style.zoom;
 
-		length = candidates.length;
-		bestCandidate = candidates[ length - 1 ];
+		// 	if (WebkitBackfaceVisibility) {
+		// 		style.zoom = ".999";
 
-		for ( var i = 0; i < length; i++ ) {
-			candidate = candidates[ i ];
-			if ( candidate.resolution >= pf.getDpr() ) {
-				bestCandidate = candidate;
-				break;
-			}
-		}
+		// 		WebkitBackfaceVisibility = picImg.offsetWidth;
 
-		if ( bestCandidate ) {
+		// 		style.zoom = currentZoom;
+		// 	}
+		// };
 
-			bestCandidate.url = pf.makeUrl( bestCandidate.url );
+		pf.setIntrinsicSize = (function() {
+			var urlCache = {};
+			var setSize = function( picImg, width, res ) {
+				fastdom.write( function() {
+					picImg.setAttribute( "width", parseInt(width / res, 10) );
```

Comment:
I've already removed this from the `frontend` version, as it's buggy and sets incorrect widths.

We also don't need it to set the width. It's weird and overrides CSS. Why does it do it to begin with?!